action/selection/open_spiel/database.cc

```diff
@@ -483,28 +552,19 @@ std::vector<double> DatabaseState::Returns() const {
 
   // Simulate the game trajectory.
   {
-    // Apply all the tuning actions. This assumes that actions take zero time.
-    for (const auto &player_action : history_) {
-      // Apply the tuning action.
-      {
-        const TuningAction &action = tuner->GetAction(player_action.action);
-        std::string action_sql = action.sql_;
-        // If hypopg is enabled and this is a CREATE INDEX action, we need to modify the action SQL.
-        // TODO(WAN): Be more robust about cReaTE inDeX.
-        bool is_create_index = action_sql.find("CREATE INDEX") == 0 || action_sql.find("create index") == 0;
-        if (use_hypopg && is_create_index) {
-          // hypopg_create_index() is ok with the action_sql ending with a semicolon.
-          action_sql = absl::StrCat("SELECT * FROM hypopg_create_index('", action_sql, "');");
-        }
-        pqxx::result rset{txn.exec(action_sql)};
-      }
-    }
+    ApplyHistory(txn, true /*use_hypopg*/);
```

Comment:
In justification of my above comment on ApplyHistory, I think this breaks the non-hypopg code path...

```suggestion
    ApplyHistory(txn, use_hypopg);
```

is what you wanted, but I think

```suggestion
    ApplyHistory(txn);
```

is what it should be refactored into.